### PR TITLE
Only ascii alphanumeric strings in tokens after v1

### PR DESCRIPTION
**Pull request by azarovh**

### CODE REVIEW

File: chainstate/src/detail/chainstateref/mod.rs
Comment (TheQuantumPhysicist) [inline chainstate/src/detail/chainstateref/mod.rs:746]: Please load the block height in here. I think this is a reasonable proposition because later, after the fork, it'll be easy to remove it.

File: chainstate/src/detail/tokens/check_utils.rs
Comment (iljakuklic) [inline chainstate/src/detail/tokens/check_utils.rs:33]: Two points here:
* Can use `Result::is_ok_and` but that has been stabilised fairly recently, not sure we want to require such a recent compiler version
* `is_ascii_alphanumeric` may be somewhat restrictive for description for exmaple. It would reject anything with a space or usual punctuation `.,;:!?` in it. I think ticker and name need a separate set of valid characters different from description.

Comment (iljakuklic) [reply chainstate/src/detail/tokens/check_utils.rs:33]: Btw there's `std::str::from_utf8` that lets you convert `&[u8]` to `&str` without allocating a `String` in a validating way.

File: test-utils/src/lib.rs
Comment (ImplOfAnImpl) [inline test-utils/src/lib.rs:180]: It's better to rename `random_string` to suggest that it can only produce alphanum characters

**Review by ImplOfAnImpl — APPROVED** [review]

File: chainstate/src/detail/tokens/check_utils.rs
Comment (azarovh) [reply chainstate/src/detail/tokens/check_utils.rs:33]: > is_ascii_alphanumeric may be somewhat restrictive for description for exmaple. 

that's by design to avoid consequences that we don't understand at the moment. Can easily be changed later

> Btw there's std::str::from_utf8

nice thanks

File: chainstate/src/detail/chainstateref/mod.rs
Comment (azarovh) [reply chainstate/src/detail/chainstateref/mod.rs]: done

File: chainstate/src/detail/tokens/check_utils.rs
Comment (azarovh) [reply chainstate/src/detail/tokens/check_utils.rs]: also simplified `check_is_text_ascii_alphanumeric`

Comment (iljakuklic) [reply chainstate/src/detail/tokens/check_utils.rs]: > Can easily be changed later

Is it not a part of consensus rules?

**Comment by TheQuantumPhysicist** [comment]
Please don't merge this until I slap another approval. I still have to review it again. I have some doubts.

**Review by TheQuantumPhysicist — CHANGES_REQUESTED** [review]
Just to ensure it won't be merged by mistake.

File: chainstate/src/detail/tokens/check_utils.rs
Comment (azarovh) [reply chainstate/src/detail/tokens/check_utils.rs]: I mean with a fork

File: chainstate/src/detail/chainstateref/mod.rs
Comment (TheQuantumPhysicist) [inline chainstate/src/detail/chainstateref/mod.rs:750]: Better use the error PrevBlockIndexNotFound. Also, this is not a non-chalant error... this is an invariant broken. 

Why not use the function that was made for this anyway? `get_previous_block_index_for_check_block()`

File: test-utils/src/lib.rs
Comment (TheQuantumPhysicist) [inline test-utils/src/lib.rs:49]: It's actually much less... because half of u8 range is printable characters, not necessarily alphanumeric.

**Review by TheQuantumPhysicist — APPROVED** [review]
Besides that minor thing, I think this is fine.

File: chainstate/src/detail/chainstateref/mod.rs
Comment (azarovh) [reply chainstate/src/detail/chainstateref/mod.rs]: > Better use the error PrevBlockIndexNotFound. 

Agree, done.

> Why not use the function that was made for this anyway?

It returns incompatible Error type. It doesn't do much so I think it's ok to leave it like this.

**Review by muursh — APPROVED** [review]

**Pull request merged**
